src/DynamoCoreWpf/Utilities/GroupStyleItemSelector.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Dynamo.Graph.Annotations;
+using System.Windows;
+using System.Windows.Controls;
+
+namespace Dynamo.Wpf.Utilities
+{
+    /// <summary>
+    /// This class will be used for select a specific Style depending if the list item is a Separator or MenuItem
+    /// </summary>
+    public class GroupStyleItemSelector : StyleSelector
+    {
+        public override Style SelectStyle(object item, DependencyObject container)
+        {
+            if (item is GroupStyleSeparator)
+            {
+                return (Style)((FrameworkElement)container).FindResource("GroupStyleSeparatorStyle");
```

Comment:
Will this resource will be found always if the item is GroupStyleSeparator?

Reply:
@reddyashish yes, the GroupStyleSeparatorStyle resource was added as a style in AnnotationView.xaml, then it should always exists.